text/0035-opentelemetry-protocol.md

```diff
@@ -0,0 +1,425 @@
+
+- Allows backpressure signalling from telemetry data destinations to sources. This is important for implementing reliable multi-hop telemetry data delivery all the way from the source to the destination via intermediate nodes, each having different processing capacity and thus requiring different data transfer rates.
+
+## Protocol Details
+
+OTLP defines the encoding of telemetry data and the protocol used to exchange data between the client and the server.
+
+This specification defines how OTLP is implemented over [gRPC](https://grpc.io/) and specifies corresponding [Protocol Buffers](https://developers.google.com/protocol-buffers/docs/overview) schema. Future extensions to OTLP may define implementations over other transports. For details of gRPC service definition see section [gRPC Transport](#grpc-transport).
+
+OTLP is a request/response style protocols: the clients send requests, the server replies with corresponding responses. This document defines one requests and response type: `Export`.
+
+### Export Request and Response
+
+After establishing the underlying transport the client starts sending telemetry data using `Export` requests.  The client continuously sends a sequence of `Export` requests to the server and expects to receive a response to each request:
+
+![Request-Response](images/otlp-request-response.png)
+
+_Note: this protocol is concerned with reliability of delivery between one pair of client/server nodes and aims to ensure that no data is lost in-transit between the client and the server. Many telemetry collection systems have intermediary nodes that the data must travel across until reaching the final destination (e.g. application -> agent -> collector -> backend). End-to-end delivery guarantees in such systems is outside of the scope of OTLP. The acknowledgements described in this protocol happen between a single client/server pair and do not span intermediary nodes in multi-hop delivery paths._
+
+#### OTLP over gRPC
+
+For gRPC transport OTLP uses Unary RPC to send export requests and receives responses.
+
+After sending the request the client MAY wait until the response is received from the server. In that case there will be at most only one request in flight that is not yet acknowledged by the server.
+
+![Unary](images/otlp-sequential.png)
+
+Sequential operation is recommended when simplicity of implementation is desirable and when the client and the server are connected via very low-latency network, such as for example when the client is an instrumented application and the server is a OpenTelemetry Service running as a local daemon.
+
+The implementations that need to achieve high throughput SHOULD support concurrent Unary calls to achieve higher throughput. The client SHOULD send new requests without waiting for the response to the earlier sent requests, essentially creating a pipeline of requests that are currently in flight that are not acknowledged.
+
+![Streaming](images/otlp-concurrent.png)
```

Comment:
this picture implies ordered responses. Is it intentional or just to not over-complicate the picture?

Reply:
Just to keep it simple. The responses may arrive out of order but it has no implications on the implementation. The protocol does not rely in any way in the ordering of responses.